lib/sidekiq.rb

```diff
@@ -117,25 +124,33 @@ def self.redis_info
       else
         conn.info
       end
-    rescue Redis::CommandError => ex
+    rescue command_error => ex
       # 2850 return fake version when INFO command has (probably) been renamed
       raise unless /unknown command/.match?(ex.message)
       FAKE_INFO
     end
   end
 
   def self.redis_pool
-    @redis ||= Sidekiq::RedisConnection.create
+    @redis ||= connection_adapter.create
   end
 
   def self.redis=(hash)
     @redis = if hash.is_a?(ConnectionPool)
       hash
     else
-      Sidekiq::RedisConnection.create(hash)
+      connection_adapter.create(hash)
     end
   end
 
+  def self.base_redis_error
```

Comment:
I don't really want to see this low-level Redis stuff mixed into Sidekiq's top-level Module. Can this go into Sidekiq::Redis* somewhere?

Reply:
I replaced it by `connection_adapter::BaseError` for simplicity. 

But if you find this weird I can do a nested module.